File: download_images.py
"""
Download high-quality dry fruits images
Using requests library for better HTTP handling
"""

import requests
import os
from pathlib import Path

# Define image URLs from reliable sources
images = {
    "cashews.jpg": "https://source.unsplash.com/featured/1200x800/?cashews,nuts",
    "dates-black.jpg": "https://source.unsplash.com/featured/1200x800/?dates,fruit,dark",
    "dates-white.jpg": "https://source.unsplash.com/featured/1200x800/?dates,golden",
    "walnuts.jpg": "https://source.unsplash.com/featured/1200x800/?walnuts",
    "raisins-black.jpg": "https://source.unsplash.com/featured/1200x800/?raisins,dark",
    "raisins-golden.jpg": "https://source.unsplash.com/featured/1200x800/?raisins,golden",
    "apricots.jpg": "https://source.unsplash.com/featured/1200x800/?apricots,dried",
    "figs.jpg": "https://source.unsplash.com/featured/1200x800/?figs,dried"
}

public_dir = Path("public")
public_dir.mkdir(exist_ok=True)

headers = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
}

print("🌟 Downloading dry fruits images from Unsplash...\n")

success = 0
failed = 0

for filename, url in images.items():
    print(f"📥 Downloading {filename}...")
    try:
        response = requests.get(url, headers=headers, timeout=30, allow_redirects=True)
        response.raise_for_status()
        
        filepath = public_dir / filename
        filepath.write_bytes(response.content)
        
        size_kb = len(response.content) / 1024
        print(f"   ✅ Success - {size_kb:.2f} KB\n")
        success += 1
        
    except Exception as e:
        print(f"   ❌ Failed - {str(e)}\n")
        failed += 1

print(f"\n📊 Download Summary:")
print(f"   ✅ Success: {success}/{len(images)}")
print(f"   ❌ Failed: {failed}/{len(images)}")

if success == len(images):
    print(f"\n🎉 All images downloaded successfully!")

File: download_images.js

import fs from 'fs';
import https from 'https';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const images = [
    // Oils
    { url: 'https://upload.wikimedia.org/wikipedia/commons/e/e3/Bottle%2C_castor_oil_%28AM_1969.210-5%29.jpg', dest: 'castor-oil.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/2/29/Peanut_oil_bottle.jpg', dest: 'groundnut-oil.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/d/d3/Mustard_Oil.jpg', dest: 'mustard-oil.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/8/82/Bottle_1_liter_Sunflower_refined_oil.jpg', dest: 'sunflower-oil.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/0/07/SweetAlmondOil.png', dest: 'almond-oil.png' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/6/6d/Safflower_oil.jpg', dest: 'safflower-oil.jpg' },

    // Dry Fruits
    { url: 'https://upload.wikimedia.org/wikipedia/commons/6/69/Almonds_and_nnuts.JPG', dest: 'almonds-new.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/3/30/Young_cashew_nuts.jpg', dest: 'cashews-new.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/d/d3/Date_fruit.jpg', dest: 'dates-black.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/6/6e/Ripe_and_dry_dates_fruit_bunches.jpg', dest: 'dates-white.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/2/22/Walnuts_-_whole_and_open_with_halved_kernel.jpg', dest: 'walnuts.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/5/5a/BLACK_RAISINS.jpg', dest: 'raisins-black.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/f/ff/Golden_Jumbo_Raisins_comparison.jpg', dest: 'raisins-golden.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/d/d8/Dried_apricots.jpg', dest: 'apricots.jpg' },
    { url: 'https://upload.wikimedia.org/wikipedia/commons/1/13/Dried_figs_%288407322626%29.jpg', dest: 'figs.jpg' },
];

const downloadImage = (url, dest) => {
    const filePath = path.join(__dirname, 'public', dest);
    const file = fs.createWriteStream(filePath);

    https.get(url, { headers: { 'User-Agent': 'Mozilla/5.0' } }, (response) => {
        if (response.statusCode !== 200) {
            console.error(`Failed to download ${url}: Status Code ${response.statusCode}`);
            return;
        }
        response.pipe(file);
        file.on('finish', () => {
            file.close();
            console.log(`Downloaded ${dest}`);
        });
    }).on('error', (err) => {
        fs.unlink(filePath, () => { }); // Delete the file async. (But we don't check result)
        console.error(`Error downloading ${url}: ${err.message}`);
    });
};

// Ensure public dir exists (it should)
if (!fs.existsSync(path.join(__dirname, 'public'))) {
    console.log("Public directory not found, checking current dir...");
}

images.forEach(img => downloadImage(img.url, img.dest));
